packages/protocol/contracts/dapis/rrpDapi.sol

```diff
@@ -2,13 +2,91 @@
 pragma solidity 0.8.4;
 
 import "./reducers/Reducer.sol";
+import "./rrpDapiServer.sol";
 
 // A stub for an RRP dAPI contract
 // When the user wants to deploy a dAPI contract, they inherit the specific
 // reducer contract such as:
 //       contract myRrpDapi is rrpDapi, Median {
 //           // Any specific implementation required
 //       }
-abstract contract rrpDapi is Reducer {
-    // TODO: Implement
+abstract contract RrpDapi is Reducer {
+
+    RrpDapiServer public rrpDapiServer;
+
+    uint256 public immutable maxAnswersInStorage;
```

Comment:
This is not practical. If I'm a prediction market, I don't want any kind of upper limit to my active requests if I can avoid it. This becomes for of a problem when you're overwriting individual responses in `RrpDapiServer` because then you would be sharing this limit with everyone else.

Reply:
Why not? You can estimate how many requests you'll make in, say, a day, and then use that to create a limit (if desired).

Reply:
Also, there is the possibility for an expiry date implementation. But I think it has similar usefulness but is more complicated to implement.

Reply:
What if you need to exceed the limit, you close up shop? That's not nice. I think expiry date is required and it's complexity would be justified if it was being used to overwrite individual reponses and not only the reduced response.